### PR TITLE
[SYCLomatic] Support migration of cub::DeviceSpmv::CsrMV

### PR DESCRIPTION
**Pull request by yihwang-nv**
E2E test https://github.com/oneapi-src/SYCLomatic-test/pull/721.

### CODE REVIEW

**Review by zhiweij1 — APPROVED** [review]

File: clang/lib/DPCT/Rewriters/CUB/RewriterDeviceSpmv.cpp [hunk 1]
```diff
@@ -0,0 +1,37 @@
+//===--------------- RewriterDeviceSelect.cpp -----------------------------===//
```
Comment (tangjj11) [inline clang/lib/DPCT/Rewriters/CUB/RewriterDeviceSpmv.cpp]:
```suggestion
//===--------------- RewriterDeviceSpmv.cpp -----------------------------===//
```

File: clang/runtime/dpct-rt/include/dpct/sparse_utils.hpp [hunk 2]
```diff
@@ -163,6 +163,30 @@ void csrmv(sycl::queue &queue, oneapi::mkl::transpose trans, int num_rows,
                           row_ptr, col_ind, x, beta, y);
 }
 
+/// Computes a CSR format sparse matrix-dense vector product. y = A * x
+///
+/// \param [in] queue The queue where the routine should be executed. It must
+/// have the in_order property when using the USM mode.
+/// \param [in] num_rows Number of rows of the matrix A.
```
Comment (tangjj11) [inline clang/runtime/dpct-rt/include/dpct/sparse_utils.hpp:176]: the param order is wrong

**Review by tangjj11 — CHANGES_REQUESTED** [review]

File: clang/lib/DPCT/Rewriters/CUB/RewriterDeviceSpmv.cpp (hunk 1 above)
Comment (yihwang-nv) [reply clang/lib/DPCT/Rewriters/CUB/RewriterDeviceSpmv.cpp]: fixed

File: clang/runtime/dpct-rt/include/dpct/sparse_utils.hpp (hunk 2 above)
Comment (yihwang-nv) [reply clang/runtime/dpct-rt/include/dpct/sparse_utils.hpp:176]: fixed

**Review by tangjj11 — APPROVED** [review]

**Review by zhimingwang36 — APPROVED** [review]

**Pull request merged**